### PR TITLE
Add API to Glossary

### PR DESCRIPTION
**Pull request by rootqa**
Add API and DAG information

### CODE REVIEW

**Comment by minimalsm** [comment]
Thanks for the PR @rootqa 🙏  

@samajammin @wackerow any thoughts as to adding these terms to the glossary?

My thoughts are:

#### DAG
- We don't use the term DAG anywhere on the site. 
- Combine this with the merge happening in ~3/4 months and I don't think it makes sense to add

#### API
- Don't mind adding API
> Is the term/definition directly relevant to Ethereum?
- Only negative for me is that it is not specific to Ethereum as per listing criteria above 
- Could be useful if implemented the on hover glossary tooltip

**Comment by rootqa** [comment]
> Thanks for the PR @rootqa 🙏
> 
> @samajammin @wackerow any thoughts as to adding these terms to the glossary?
> 
> My thoughts are:
> 
> #### DAG
> * We don't use the term DAG anywhere on the site.
> * Combine this with the merge happening in ~3/4 months and I don't think it makes sense to add
> 
> #### API
> * Don't mind adding API
> 
> > Is the term/definition directly relevant to Ethereum?
> 
> * Only negative for me is that it is not specific to Ethereum as per listing criteria above
> * Could be useful if implemented the on hover glossary tooltip

hey @minimalsm  thanks for interest so actually geth.ethereum.org and Developer Resources use a DAG and API and i think a lot of new developers are interested in that. thanks again!

File: src/content/glossary/index.md [hunk 1]
```diff
@@ -28,6 +28,10 @@ An object containing an [address](#address), balance, [nonce](#nonce), and optio
 
 Most generally, this represents an [EOA](#eoa) or [contract](#contract-accouint) that can receive (destination address) or send (source address) [transactions](#transaction) on the blockchain. More specifically, it is the rightmost 160 bits of a [Keccak hash](#keccak-256) of an [ECDSA](#ecdsa) [public key](#public-key).
 
+### application programming interface
+
+An API is a set of definitions and protocols for building and integrating application software. API stands for application programming interface.
```
Comment (corwintines) [inline src/content/glossary/index.md:33]:
```suggestion
An Application Programming Interface (API) is a set of definitions for how to use a piece of software. An API sits between an application and a web server, and facilitates the transfer of data between them.
```

File: src/content/glossary/index.md [hunk 2]
```diff
@@ -123,6 +127,10 @@ A crosslink provides a summary of a shard's state. It's how [shard](#shard) chai
 
 ## D {#section-d}
 
+### DAG
+
+DAG stands for Directed Acyclic Graph and without going into too many details, it is a crucial file in ETH and ETC mining. DAG file size grows over time, exactly every 30.000 blocks (Ethereum) or every 100 hours to be precise.
```
Comment (corwintines) [inline src/content/glossary/index.md:132]:
```suggestion
A Directed Acyclic Graph (DAG) is used to make the Ethash Proof-of-Work (PoW) ASIC resistant. The DAG dataset is a resource used when calculating the proof-of-work, and requires choosing subsets of a fixed resource dependent on the nonce and block header. A new DAG dataset is created every 30000 blocks (a 100 hour window, called an epoch), and is a few GB in size.
```
This means that calculating the PoW requires choosing subsets of a fixed resource dependent on the nonce and block header.

**Review by corwintines — COMMENTED** [review]
I think these are relevant terms, but leaving some suggestions on how to define them.

**Comment by minimalsm** [comment]
Hey @rootqa thanks again for the PR.

I chatted over this with the team and confirmed because of the lack of usage and the fact we're moving to proof-of-stake in the very near future we don't think adding DAG adds value to the glossary.

We agreed that adding API is a nice addition because it clarifies the API vs ABI confusion and makes the term more approachable for people who may not be developers.

Merging the changes now :] 

Be sure to [join the discord](https://discord.gg/CetY6Y4) if you are interested in contributing further to the project or have any questions for the team. And remember to [claim your POAP](https://ethereum.org/en/contributing/#poap) also 🥳!

**Comment by minimalsm** [comment]
@all-contributors please add @rootqa for docs

**Pull request merged**